modules/cnv_install_1_4.adoc

```diff
@@ -0,0 +1,64 @@
+[[install_cnv_apb]]
+=== Installing {ProductName} to {product-title}
+
+The `kubevirt-ansible` RPM contains the latest automation for deploying 
+{ProductName}. This procedure installs all {ProductName} components to your 
+{product-title} cluster. 
+
+This procedure installs the following components:
+
+* {ProductName} core components (KubeVirt)
+* Containerized data importer (CDI) controller
+* Multus, Open vSwitch (OVS), and SR-IOV container network interface plug-ins
+* Updated {product-title} web console
+
+.Prerequisites
+* {product-title} 3.11 cluster, running
+* User with `cluster-admin` privileges
+
+.Procedure
+
+. Log into the {product-title} cluster as an admin user:
++
+----
+$ oc login -u system:admin
+----
+
+. Install the `kubevirt-ansible` RPM and its dependencies:
++
+----
+$ yum install kubevirt-ansible
+----
+
+. If your cluster has unique requirements, customize the 
```

Comment:
What am I supposed to customize in this file? I'd say something like "If your cluster \<requires or uses\> \<things you can customize here\>, modify the `/usr/share/ansible/kubevirt-ansible/vars/cnv.yml` file."

Reply:
I wonder, is there an option to use environment variables instead of copying the files?

Reply:
I don't think you can use env variables (it does not show on https://docs.ansible.com/ansible/latest/user_guide/playbooks_variables.html#variable-precedence-where-should-i-put-a-variable )

but I'm with you on not copying cnv.yml AT ALL. We should just say:

In order to use a custom repository instead ofregistry.access.redhat.com, add `-e registry_ur=registry.example.com -e cnv_repo_tag=local-repo-tag-for-cnv` to the command line.

To test the still-unsupported SR-IOV add `-e +deploy_sriov_plugin=true`

Reply:
using `-e xxxx=yyy` has highest precedence;  it's `extra vars`.

Reply:
> using `-e xxxx=yyy` has highest precedence; it's `extra vars`.

correct. that's why I want users to use it to overwrite values in the chipped all.yml and cnv.yml.

Reply:
To close the loop on this discussion, I think we all agree that we want to minimize the problem space by lowering the amount of "supported customizations".  Using extra vars will avoid copying any files and give the user the flexibility needed to make kubevirt-anisble work for their environment.  With that being said, can I get an ack from folks on the __minimal__ supported extra vars?
@simon3z @ohadlevy @dankenigsberg 

```
# Container Registry String: docker_prefix:docker_tag
docker_prefix
docker_tag
```

Reply:
This is the complete list. Product management may ask you change it in the future. Even *I* may ask you to include deploy_sriov to this list in a near future. But currently this is it. Only two extra args are supported, and only for offline deployment.

Reply:
Adding one more...
```yaml
# Container Registry String: docker_prefix:docker_tag
docker_prefix
docker_tag

apb_action
```

Reply:
I don't know what `apb_action` does. This should not be a verbatim table, just a plain paragraph telling in order to do X run with `-e apb_action=bla`.

Reply:
`apb_action=provision` and `apb_acton=deprovision`.  `apb_acton=deprovision` will be needed for deletion.

Reply:
@rthallisey @dankenigsberg 

1. Can you clarify if registry_url and cnv_repo_tag are supported args, or only docker_prefix, docker_tag and apb_action? 

2. Can you give an example of why someone would want to change docker_prefix and docker_tag?

Thanks!

Reply:
> Can you clarify if registry_url and cnv_repo_tag are supported args, or only docker_prefix, docker_tag and apb_action?
>
I would stick with just `docker_prefix`, `docker_tag` and `apb_action` because `registry_url` and `cnv_repo_tag` are both covered by `docker_prefix`;  it simplifies things.

> Can you give an example of why someone would want to change docker_prefix and docker_tag?
>
If a user has an on premise image registry where they hold our images, the install will need to pull images from their internal registry rather than Red Hat's registry.

Reply:
> > Can you clarify if registry_url and cnv_repo_tag are supported args, or only docker_prefix, docker_tag and apb_action?
> 
> I would stick with just `docker_prefix`, `docker_tag` and `apb_action` because `registry_url` and `cnv_repo_tag` are both covered by `docker_prefix`; it simplifies things.

Have you tested your suggestion, @rthallisey ? I see that our code uses `registry_url` directly in several places.

> 
> > Can you give an example of why someone would want to change docker_prefix and docker_tag?
> 
> If a user has an on premise image registry where they hold our images, the install will need to pull images from their internal registry rather than Red Hat's registry.

this example does not explain why a customer in his right mind would like to change `docker_tag` from its sane default `v1.4.0`.

May I repeat my request to start with my text, and only fix errors in it - if spotted? my text desribes what cnv.yml does to differentiate from kubevirt. it is also what QE does to test from brew. Let us write it down and get going.

Reply:
Done, @dankenigsberg - check out the latest commit and let me know what I got wrong. :)

Reply:
The web_ui doesn't use docker_prefix.  I must've missed it in my review.  Also, @dankenigsberg looks like network hardcodes the registry to upstream images.  I think these are both bugs @nellyc.
```yaml
multus_provisioner_repo: "docker.io/nfvpe/multus"
cni_provisioner_repo: "quay.io/schseba/cni-plugins"
ovs_provisioner_repo: "quay.io/kubevirt/ovs-cni-plugin"
```

Reply:
ignore my comment.  cnv.yaml covers it.